tests/command/test_reword.rs

```diff
@@ -0,0 +1,297 @@
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+    git.init_repo()?;
+    git.commit_file("test1", 1)?;
+    git.run(&["branch", "test1"])?;
+    git.commit_file("test2", 2)?;
+
+    let (stdout, _stderr) = git.run(&["smartlog"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    :
+    O 62fc20d2 (test1) create test1.txt
+    |
+    @ 96d1c37a (> master) create test2.txt
+    "###);
+
+    git.run(&["reword", "--message", "foo"])?;
+
+    let (stdout, _stderr) = git.run(&["smartlog"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    :
+    O 62fc20d2 (test1) create test1.txt
+    |
+    @ c1f5400a (> master) foo
+    "###);
+
+    Ok(())
+}
+
+#[test]
+fn test_reword_current_commit_not_head() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+    git.init_repo()?;
+    git.commit_file("test1", 1)?;
+    git.run(&["branch", "test1"])?;
+    git.commit_file("test2", 2)?;
+    git.run(&["prev"])?;
+
+    let (stdout, _stderr) = git.run(&["smartlog"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    :
+    @ 62fc20d2 (test1) create test1.txt
+    |
+    O 96d1c37a (master) create test2.txt
+    "###);
+
+    git.run(&["reword", "--message", "foo"])?;
+
+    let (stdout, _stderr) = git.run(&["smartlog"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    :
+    @ a6f88684 (test1) foo
+    |
+    O 5207ad50 (master) create test2.txt
+    "###);
+
+    Ok(())
+}
+
+#[test]
+fn test_reword_with_multiple_messages() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+    git.init_repo()?;
+    git.commit_file("test1", 1)?;
+
+    let (stdout, _stderr) = git.run(&["log", "-n", "1", "--format=%h%n%B"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    62fc20d
+    create test1.txt
+    "###);
+
+    git.run(&["reword", "-m", "foo", "-m", "bar"])?;
+
+    let (stdout, _stderr) = git.run(&["log", "-n", "1", "--format=%h%n%B"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    34ae21e
+    foo
+
+    bar
+
+    "###);
+
+    Ok(())
+}
+
+#[test]
+fn test_reword_removes_comment_lines_from_messages() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+    git.init_repo()?;
+    git.commit_file("test1", 1)?;
+
+    let (stdout, _stderr) = git.run(&["log", "-n", "1", "--format=%h%n%B"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    62fc20d
+    create test1.txt
+    "###);
+
+    // try adding several messages that start w/ '#'
+    git.run(&["reword", "-m", "foo", "-m", "# bar", "-m", "#", "-m", "buz"])?;
+
+    // confirm the '#' messages aren't present
+    let (stdout, _stderr) = git.run(&["log", "-n", "1", "--format=%h%n%B"])?;
+    insta::assert_snapshot!(stdout, @r###"
+    2666020
+    foo
+
+    buz
```

Comment:
I just added this test today, to test that we're removing comment lines. See other comment about how comments are handled, though.

Reply:
I updated this to confirm that we *are* allowing "comments" from the CLI.